include/RAJA/policy/hip/launch.hpp

```diff
@@ -239,6 +263,70 @@ struct LaunchExecute<RAJA::policy::hip::hip_launch_t<async, nthreads>>
   }
 };
 
+/*
+  Loop methods which rely on a copy of threaIdx/BlockDim
+  for performance. In collaboration with AMD we have have this
+  to be more performant.
+*/
+
+namespace expt
+{
+
+template<named_dim DIM>
+struct hip_ctx_thread_loop;
+
+using hip_ctx_thread_loop_x = hip_ctx_thread_loop<named_dim::x>;
+using hip_ctx_thread_loop_y = hip_ctx_thread_loop<named_dim::y>;
+using hip_ctx_thread_loop_z = hip_ctx_thread_loop<named_dim::z>;
+
+template<named_dim DIM, typename Dim3Like>
+RAJA_INLINE RAJA_DEVICE int get_dim(Dim3Like const& d)
+{
+  if constexpr (DIM == named_dim::x)
```

Comment:
@MrBurmark @johnbowen42 , is this the usage of `if constexpr` you all had in mind?